src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java

```diff
@@ -45,7 +48,38 @@ public ClientCommandBuilder(String server, String project, boolean skipTLSVerify
             return n;
         }
         for (Object o : l) {
-            n.add(o.toString());
+            String s = o.toString();
+            if (s.contains(" ")) {
+                // the Jenkins QuotedStringTokenizer is not quite 
+                // copacetic with template parameters defined in 
+                // groovy variables (specifically strings where you have spaces but don't have 
+                // to encapsulate with " or ').
+                
+                // that said, for now at least, we'll try to use the Jenkins
+                // QuotedStringTokenizer for the other " and ' scenarios, as well
+                // as adjusting input to it with the logic here so that it 
+                // properly handles the groovy string vars with spaces scenario 
+                // correctly.
+                
+                // So, if the arg that comes in either 
+                // a) has neither ' or "", or
+                // b) has ", but doesn not start with it, and does not start with '
+                // then wrapper with '
+                if ((!s.contains("'") && !s.contains("\"")) ||
+                   (!s.startsWith("\"") && !s.startsWith("'") && s.contains("\"")))
+                    s = "'" + s + "'";
+                // if the arg that comes in either 
+                // c) has ', but does not start with it, and does not start with "
+                // then wrapper with "
+                if (!s.startsWith("\"") && !s.startsWith("'") && s.contains(("'")))
```

Comment:
you are doing this check against the string you just (potentially) added quotes to.  intentional?

Reply:
ahh ... thx, will address